src/generators/operationInterfaceGenerator.ts

```diff
@@ -0,0 +1,240 @@
+  getOperationResponseType
+} from "./utils/responseTypeUtils";
+import { generateOperationJSDoc } from "./utils/docsUtils";
+import { addTracingOperationImports } from "./utils/tracingUtils";
+import {
+  addPagingEsNextRef,
+  addPagingImports,
+  preparePageableOperations,
+  writeAsyncIterators
+} from "./utils/pagingOperations";
+
+/**
+ * Function that writes the code for all the operations.
+ * It will generate one file per operation group and each file contains:
+ *    - A class definition for the operation group
+ *    - Methods and overrides for each operation
+ *    - OperationSpecs for each operation
+ * @param clientDetails client details
+ * @param project project for code generation
+ */
+export function generateOperationsInterfaces(
+  clientDetails: ClientDetails,
+  project: Project
+): void {
+  let fileNames: string[] = [];
+
+  // Toplevel operations are inlined in the client
+  const operationGroups = clientDetails.operationGroups.filter(
+    og => !og.isTopLevel
+  );
+
+  operationGroups.forEach(operationDetails => {
+    fileNames.push(normalizeName(operationDetails.name, NameType.File));
+    generateOperation(operationDetails, clientDetails, project);
+  });
+
+  if (operationGroups.length) {
+    const operationIndexFile = project.createSourceFile(
+      `${clientDetails.srcPath}/operationsInterfaces/index.ts`,
+      undefined,
+      { overwrite: true }
+    );
+
+    operationIndexFile.addExportDeclarations(
+      fileNames.map(fileName => {
+        return {
+          moduleSpecifier: `./${fileName}`
+        } as ExportDeclarationStructure;
+      })
+    );
+  }
+}
+
+/**
+ * This function creates a file for a given Operation Group
+ */
+function generateOperation(
+  operationGroupDetails: OperationGroupDetails,
+  clientDetails: ClientDetails,
+  project: Project
+): void {
+  const name = normalizeName(operationGroupDetails.name, NameType.File);
+  const operationInterfaceGroupFile = project.createSourceFile(
+    `${clientDetails.srcPath}/operationsInterfaces/${name}.ts`,
+    undefined,
+    { overwrite: true }
+  );
+
+  addImports(operationGroupDetails, operationInterfaceGroupFile, clientDetails);
+  addInterface(
+    operationInterfaceGroupFile,
+    operationGroupDetails,
+    clientDetails
+  );
+}
+
+function getReturnType(
+  operation: OperationDetails,
+  importedModels: Set<string>,
+  modelNames: Set<string>
+) {
+  const responseName = getOperationResponseType(
+    operation,
+    importedModels,
+    modelNames
+  );
+
+  return operation.isLRO
+    ? `Promise<LROPoller<${responseName}>>`
+    : `Promise<${responseName}>`;
+}
+
+/**
+ * Adds an Operation group class to the generated file
+ */
+function addInterface(
+  operationGroupFile: SourceFile,
+  operationGroupDetails: OperationGroupDetails,
+  clientDetails: ClientDetails
+) {
+  let importedModels = new Set<string>();
+
+  let allModelsNames = getAllModelsNames(clientDetails);
+
+  const className = normalizeName(
```

Comment:
`className` needs to be renamed to `interfaceName`.

Reply:
Done (in the next commit)